### PR TITLE
ScreenshotUtils Mappings

### PR DESCRIPTION
**Pull request by ramidzkh**
Simple mappings for ScreenshotUtils

### CODE REVIEW

File: mappings/net/minecraft/client/texture/NativeImage.mapping [hunk 1]
```diff
@@ -77,6 +77,8 @@ CLASS cub net/minecraft/client/texture/NativeImage
 		ARG 7 shiftX
 		ARG 8 shiftY
 		ARG 10 glyph
+	METHOD a readPixels (Z)V
```
Comment (asiekierka) [inline mappings/net/minecraft/client/texture/NativeImage.mapping]: Can we come up with a better name for this?

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 2]
```diff
@@ -1,5 +1,22 @@
 CLASS cvv net/minecraft/client/util/ScreenshotUtils
 	FIELD a LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD b DATE_FORMAT Ljava/text/DateFormat;
+	METHOD a captureFramebuffer (IILctr;)Lcub;
+		ARG 0 width
+		ARG 1 height
+		ARG 2 framebuffer
 	METHOD a getScreenshotFilename (Ljava/io/File;)Ljava/io/File;
 		ARG 0 directory
+	METHOD a captureScreenshot (Ljava/io/File;IILctr;Ljava/util/function/Consumer;)V
```
Comment (asiekierka) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]: Does this only capture, or also save?

Comment (ramidzkh) [reply mappings/net/minecraft/client/util/ScreenshotUtils.mapping]: `captureAndWriteScreenshot` seems too lengthy

Comment (asiekierka) [reply mappings/net/minecraft/client/util/ScreenshotUtils.mapping]: saveScreenshot sounds fine to me?

File: mappings/net/minecraft/client/texture/NativeImage.mapping (hunk 1 above)
Comment (ramidzkh) [reply mappings/net/minecraft/client/texture/NativeImage.mapping]: It literally reads the pixel data from the GlStateManager. `readPixelData` is my other idea.

Comment (Gegy) [reply mappings/net/minecraft/client/texture/NativeImage.mapping]: So it captures pixel data from the screen and sets it in the `NativeImage`?

Comment (ramidzkh) [reply mappings/net/minecraft/client/texture/NativeImage.mapping]: Yes.

Comment (asiekierka) [reply mappings/net/minecraft/client/texture/NativeImage.mapping]: Why not captureDisplay, or something akin?

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
 	FIELD a LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD b DATE_FORMAT Ljava/text/DateFormat;
+	METHOD a captureFramebuffer (IILctr;)Lcub;
+		ARG 0 width
+		ARG 1 height
+		ARG 2 framebuffer
 	METHOD a getScreenshotFilename (Ljava/io/File;)Ljava/io/File;
 		ARG 0 directory
+	METHOD a saveScreenshot (Ljava/io/File;IILctr;Ljava/util/function/Consumer;)V
+		ARG 0 directory
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]:
```suggestion
		ARG 0 runDir
```

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+		ARG 1 width
+		ARG 2 height
+		ARG 3 framebuffer
+		ARG 4 callback
+	METHOD a saveScreenshot (Ljava/io/File;Ljava/lang/String;IILctr;Ljava/util/function/Consumer;)V
+		ARG 0 directory
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]:
```suggestion
		ARG 0 runDir
```

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+		ARG 1 name
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]:
```suggestion
		ARG 1 fileName
```

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 6]
```diff
@@ -1,5 +1,22 @@
...
+		ARG 1 height
+		ARG 2 framebuffer
 	METHOD a getScreenshotFilename (Ljava/io/File;)Ljava/io/File;
 		ARG 0 directory
+	METHOD a saveScreenshot (Ljava/io/File;IILctr;Ljava/util/function/Consumer;)V
+		ARG 0 directory
+		ARG 1 width
+		ARG 2 height
+		ARG 3 framebuffer
+		ARG 4 callback
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]:
```suggestion
		ARG 4 messageDispatcher
```

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 7]
```diff
@@ -1,5 +1,22 @@
...
+	METHOD a saveScreenshot (Ljava/io/File;Ljava/lang/String;IILctr;Ljava/util/function/Consumer;)V
+		ARG 0 directory
+		ARG 1 name
+		ARG 2 width
+		ARG 3 height
+		ARG 4 framebuffer
+		ARG 5 callback
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]:
```suggestion
		ARG 5 messageDispatcher
```

**Review by ChloeDawn — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 8]
```diff
@@ -1,5 +1,22 @@
 CLASS cvv net/minecraft/client/util/ScreenshotUtils
 	FIELD a LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD b DATE_FORMAT Ljava/text/DateFormat;
+	METHOD a captureFramebuffer (IILctr;)Lcub;
+		ARG 0 width
+		ARG 1 height
+		ARG 2 framebuffer
 	METHOD a getScreenshotFilename (Ljava/io/File;)Ljava/io/File;
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]:
```suggestion
 	METHOD a getTimestampedFile (Ljava/io/File;)Ljava/io/File;
```

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 9]
```diff
@@ -1,5 +1,22 @@
 CLASS cvv net/minecraft/client/util/ScreenshotUtils
 	FIELD a LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD b DATE_FORMAT Ljava/text/DateFormat;
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping:3]:
```suggestion
 	FIELD b TIMESTAMP Ljava/text/DateFormat;
```

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping [hunk 10]
```diff
@@ -1,5 +1,22 @@
...
+	METHOD a captureFramebuffer (IILctr;)Lcub;
+		ARG 0 width
+		ARG 1 height
+		ARG 2 framebuffer
 	METHOD a getScreenshotFilename (Ljava/io/File;)Ljava/io/File;
 		ARG 0 directory
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/util/ScreenshotUtils.mapping]:
```suggestion
 		ARG 0 screenshotsDir
```

**Review by ChloeDawn — CHANGES_REQUESTED** [review]

**Comment by ChloeDawn** [comment]
`ScreenshotUtils` -> `Screenshots`

`ScreenshotUtils.saveScreenshot(...)` -> `Screenshots.take(...)`

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/client/util/ScreenshotUtils.mapping]: Is this the game directory or the screenshots folder?

Comment (ChloeDawn) [reply mappings/net/minecraft/client/util/ScreenshotUtils.mapping]: The game directory.

**Comment by Pyrofab** [comment]
> `ScreenshotUtils -> Screenshots`

The `Objects` naming pattern is generally used for util classes that operate on `Object` instances. Given that we do not have an actual `Screenshot` class, why not simply name it singular `Screenshot` ?

`ScreenshotUtils.saveScreenshot(...) -> Screenshot.take(...)`

**Comment by ChloeDawn** [comment]
Not all utility classes are bound to a specific object type. See Guava's `Preconditions`, representing a utility for handling the concept of preconditions, not literal `Precondition` objects. The `Screenshots` utility class handles the concept of screenshots - that being image files located in an appropriately named directory.

**Comment by Pyrofab** [comment]
I guess that makes sense.

**Comment by modmuss50** [comment]
A lot of comments on this one that havent been responded to, seems easy enough to to update. (Let me know if you need help with that)

File: mappings/net/minecraft/client/util/ScreenshotUtils.mapping (hunk 9 above)
Comment (ramidzkh) [reply mappings/net/minecraft/client/util/ScreenshotUtils.mapping:3]: `TIMESTAMP_FORMAT` instead?

**Comment by liach** [comment]
Runemoro has mapped these screenshot stuff in 19w35a. this is no longer necessary.

**Pull request closed without merging**